docker/client.go

```diff
@@ -40,6 +44,35 @@ func NewClient() Client {
 	return &dockerClient{cli}
 }
 
+func (d *dockerClient) GetContainerById(containerId string) (Container, error) {
+    containerListOptions := types.ContainerListOptions{
+        Filters: filters.NewArgs(
+             filters.Arg("id", containerId),
+         ),
+    }
+    list, err := d.cli.ContainerList(context.Background(), containerListOptions)
+    if err != nil {
+        return Container{}, err
```

Comment:
In a case of error, it would probably more sense to just return `nil`. Returning an empty struct makes more sense for arrays.